src/app/scripts/cac/routing/cac-routing-itinerary.js

```diff
@@ -107,6 +121,32 @@ CAC.Routing.Itinerary = (function ($, cartodb, L, _, moment, Geocoder, Utils) {
     }
 
     /**
+     * Helper function to get the total travel time and distance for each mode in the itinerary.
+     *
+     * @param {array} legs Legs property of OTP itinerary
+     * @return {object} Mode keys mapped to formatted and raw total distance and duration
+     */
+    function getModeSummaries(legs) {
+        // minimum length of travel via a given mode for it to show up in the summary
+        // (about a city block)
+        var MIN_MODE_LENGTH_METERS = 120;
+
+        return _.chain(legs).groupBy(function(leg) {
+            return leg.transitLeg ? 'TRANSIT' : leg.mode;
+        }).mapValues(function(modeLegs) {
+            var dist = _.map(modeLegs, 'distance').reduce(function(sum, d) { return sum + d; });
+            var time = _.map(modeLegs, 'duration').reduce(function(sum, d) { return sum + d; });
```

Comment:
Lodash, in its extensive and sometimes murky bag of tricks, has `sumBy`.  So these could be 
```
var dist = _.sumBy(modeLegs, 'distance')
var time = _.sumBy(modeLegs, 'duration')
```